### PR TITLE
Improve logic for detecting what PR tests to run

### PR DESCRIPTION
**Pull request by hsubox76**
This adds detection of other tests that should run:
1) tests in packages dependent on the changed package(s)
2) special cases - firestore tests should run on firestore emulator file setup changes

### CODE REVIEW

File: scripts/run_changed.js [hunk 1]
```diff
@@ -52,35 +52,82 @@ const alwaysRunTestPaths = [
...
+  'scripts/emulator-testing/firestore-test-runner.ts': ['packages/firestore'],
+  'scripts/emulator-testing/database-test-runner.ts': ['packages/database']
+};
+
 /**
  * Identify modified packages that require tests.
  */
 async function getChangedPackages() {
+  const packageInfo = JSON.parse(
+    (await exec('npx lerna ls --json', { cwd: root })).stdout
```
Comment (Feiyang1) [inline scripts/run_changed.js]: Can use the already installed lerna at "node_modules/.bin/lerna" to remove the networking.

File: scripts/run_changed.js [hunk 2]
```diff
@@ -52,35 +52,82 @@ const alwaysRunTestPaths = [
...
     if (match && match[1]) {
-      const pkg = require(resolve(root, match[1], 'package.json'));
-      if (pkg && pkg.scripts.test) {
-        changedPackages[match[1]] = true;
+      const changedPackage = require(resolve(root, match[1], 'package.json'));
+      if (changedPackage && changedPackage.scripts.test) {
+        // Add the package itself.
+        changedPackages[match[1]] = 'direct';
+        // Add packages that depend on it.
+        for (const package in depGraph) {
```
Comment (Feiyang1) [inline scripts/run_changed.js]: Shouldn't we run tests for depending packages even if `changedPackage.scripts.test` evaluates to false? something like:
```
if (changedPackage) {
  if (changedPackage.scripts.test) {
      changedPackages[match[1]] = 'direct';
  }

  for (const package in depGraph) {
  ...
  }
}
```

File: scripts/run_changed.js (hunk 1 above)
Comment (hsubox76) [reply scripts/run_changed.js]: npx will run the local installed version if available.

File: scripts/run_changed.js (hunk 2 above)
Comment (hsubox76) [reply scripts/run_changed.js]: Yes, fixed.

**Pull request merged**